nequip/train/trainer.py

```diff
@@ -1144,12 +1144,55 @@ def __del__(self):
         for i in range(len(logger.handlers)):
             logger.handlers.pop()
 
+    def _parse_n_train_n_val(
+        self, train_dataset_size: int, val_dataset_size: int
+    ) -> tuple[int]:
+        # parse n_train and n_val (can be ints or str with percentage):
+        n_train_n_val = []
+        for n_name in ["n_train", "n_val"]:
+            n = getattr(self, n_name)
+            if isinstance(n, str) and "%" in n:
+                dataset_size = (
+                    train_dataset_size if n_name == "n_train" else val_dataset_size
+                )
```

Comment:
Total nitpick, but maybe move this into a 
```
for n_name, total_size in (("n_train", train_dataset_size), ("n_val", val_dataset_size))
```

Reply:
Ah yes good point! The sort of stuff you can miss when first drafting. Done! ([f05b395](https://github.com/mir-group/nequip/pull/438/commits/f05b395ad9ae8f51945e692073d695748fad5c80))